opentelemetry/proto/common/v1/common.proto

```diff
@@ -21,30 +21,37 @@ option java_package = "io.opentelemetry.proto.common.v1";
 option java_outer_classname = "CommonProto";
 option go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1";
 
+// AnyValue is used to represent any type of attribute value. AnyValue may contain a
+// primitive value such as a string or integer or it may contain an arbitrary nested
+// object containing arrays, key-value lists and primitives.
+message AnyValue {
+  // The value is one of the listed fields. It is valid for all values to be unspecified
+  // in which case this AnyValue is considered to be "null".
+  oneof value {
```

Comment:
Can we avoid using oneof? It is bad for a lot of languages as well as the JSON representation.

Reply:
I have tried a lot of versions without oneof but they are all either ugly or use too much memory. Can you point to the exact problems caused by oneof? OpenCensus proto used it, so I assume it is not that bad :-)

Reply:
That was mostly a rumour I heard :)). But I couldn't find how oneof is represented as JSON. It may be good to look at generated code.

Reply:
I went ahead and applied this PR's proposal on a branch in Collector. `oneof` seems to be doing fine. I tried gRPC-gateway which uses JSON and it works fine. This is what the payload looks like (see "attributes.value"):

```json
{
    "resource_spans": [
        {
            "resource": {
                "attributes": [
                    {
                        "key": "host.hostname",
                        "value": {
                            "stringValue": "testHost"
                        }
                    }
                ]
            },
            "instrumentation_library_spans": [
                {
                    "spans": [
                        {
                            "trace_id": "W47/95gDgQPSabYzgT/GDA==",
                            "span_id": "7uGbfsPBsXM=",
                            "name": "testSpan",
                            "start_time_unix_nano": 1544712660000000000,
                            "end_time_unix_nano": 1544712661000000000,
                            "attributes": [
                                {
                                    "key": "attr1",
                                    "value": {
                                        "intValue": 55
                                    }
                                }
                            ]
                        }
                    ]
                }
            ]
        }
    ]
}
```

Reply:
@open-telemetry/javascript-approvers any concerns about using `oneof`?

Reply:
I think it looks straightforward :)